tq/transfer_queue.go

```diff
@@ -87,70 +87,101 @@ func (r *retryCounter) CanRetry(oid string) (int, bool) {
 	return count, count < r.MaxRetries
 }
 
+type Batch []Transferable
+
+func (b Batch) IsFull(max int) bool { return len(b) >= max }
```

Comment:
Since you have to supply the max as an argument I'm not sure what value this function adds?